dashboard/common/db_util.py

```diff
@@ -87,26 +88,24 @@ def get_default_user_course_id(user_id):
 
 def get_user_courses_info(username):
     logger.info(get_user_courses_info.__name__)
-    course_list = []
     course_info = []
     with django.db.connection.cursor() as cursor:
-        cursor.execute("SELECT course_id FROM user WHERE sis_name= %s", [username])
+        cursor.execute('''
+            SELECT c.id, c.name, u.enrollment_type
+            FROM user u
+            JOIN course c
+                ON u.course_id=c.id
+            WHERE u.sis_name= %s;
+        ''', [username])
         courses = cursor.fetchall()
         if courses is not None:
             for course in courses:
-                course_id = incremented_id_to_canvas_id(course[0])
-                course_list.append(int(course_id))
-    if course_list:
-        course_tuple = tuple(course_list)
-        with django.db.connection.cursor() as cursor:
-            cursor.execute("select canvas_id, name from course where canvas_id in %s",[course_tuple])
-            course_names = cursor.fetchall()
-            df = pd.DataFrame(list(course_names))
-
-            if df.size > 0:
-                df.columns = ["course_id", "course_name"]
-                course_info = df.to_dict(orient='records')
-                logger.info(f"User {username} is enrolled in these courses: {course_info}")
+                course_info.append({
+                    'course_id': int(incremented_id_to_canvas_id(course[0])),
+                    'course_name': course[1],
+                    'enrollment_type': course[2]
+                })
```

Comment:
So I still think it might be worth revisiting the PR I opened against your `instructor-admin` branch. The way it is now, you will get multiple separate objects/dictionaries for each enrollment in a course, meaning the course name and ID will be repeated in two separate objects if you are both a student and instructor. The PR I opened basically just nests the data so `enrollment_type` becomes `enrollment_types`.

Reply:
I merged the PR